Crash in test_to_bytes()

Build log: https://github.com/diofant/python-gmp/actions/runs/12724074899/job/35469862043

```
2025-01-11T12:37:15.6703780Z ============================= test session starts ==============================
2025-01-11T12:37:15.6704100Z platform darwin -- Python 3.13.0, pytest-8.3.4, pluggy-1.5.0
2025-01-11T12:37:15.6704350Z 
2025-01-11T12:37:15.6704420Z   The GNU GMP version: 6.3.0
2025-01-11T12:37:15.6704650Z   Bits per limb:       64
2025-01-11T12:37:15.6704810Z   Size of a limb:      8
2025-01-11T12:37:15.6704900Z 
2025-01-11T12:37:15.6704990Z rootdir: /Users/runner/work/python-gmp/python-gmp
2025-01-11T12:37:15.6705240Z configfile: pyproject.toml
2025-01-11T12:37:15.6705410Z plugins: hypothesis-6.123.13
2025-01-11T12:37:15.6705560Z collected 58 items
2025-01-11T12:37:15.6705650Z 
2025-01-11T12:37:18.9010600Z ../../../../../../../../../Users/runner/work/python-gmp/python-gmp/tests/test_functions.py . [  1%]
2025-01-11T12:38:31.2025350Z .s..                                                                     [  8%]
2025-01-11T12:38:43.2457510Z ../../../../../../../../../Users/runner/work/python-gmp/python-gmp/tests/test_mpq.py . [ 10%]
2025-01-11T12:39:19.0260230Z ...                                                                      [ 15%]
2025-01-11T12:39:22.2193650Z ../../../../../../../../../Users/runner/work/python-gmp/python-gmp/tests/test_mpz.py . [ 17%]
2025-01-11T12:41:39.1071850Z .............................Fatal Python error: Segmentation fault
2025-01-11T12:41:39.1072170Z 
2025-01-11T12:41:39.1072370Z Current thread 0x00000001f64ecf80 (most recent call first):
2025-01-11T12:41:39.1075100Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/junkdrawer.py", line 105 in of_length
2025-01-11T12:41:39.1076610Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/data.py", line 453 in begin
2025-01-11T12:41:39.1078380Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/data.py", line 304 in run
2025-01-11T12:41:39.1079820Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/data.py", line 371 in lazy_calculate
2025-01-11T12:41:39.1082620Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/data.py", line 470 in ir_starts
2025-01-11T12:41:39.1084060Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/data.py", line 250 in ir_start
2025-01-11T12:41:39.1085470Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/data.py", line 519 in start_example
2025-01-11T12:41:39.1086970Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/data.py", line 326 in __push
2025-01-11T12:41:39.1088450Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/data.py", line 314 in run
2025-01-11T12:41:39.1099860Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/data.py", line 371 in lazy_calculate
2025-01-11T12:41:39.1101290Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/engine.py", line 1159 in generate_mutations_from
2025-01-11T12:41:39.1103120Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/engine.py", line 1110 in generate_new_examples
2025-01-11T12:41:39.1105280Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/engine.py", line 1283 in _run
2025-01-11T12:41:39.1109530Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/internal/conjecture/engine.py", line 791 in run
2025-01-11T12:41:39.1110980Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/core.py", line 1221 in run_engine
2025-01-11T12:41:39.1112980Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/hypothesis/core.py", line 1765 in wrapped_test
2025-01-11T12:41:39.1114770Z   File "/Users/runner/work/python-gmp/python-gmp/tests/test_mpz.py", line 524 in test_to_bytes
2025-01-11T12:41:39.1116420Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/python.py", line 159 in pytest_pyfunc_call
2025-01-11T12:41:39.1117740Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_callers.py", line 103 in _multicall
2025-01-11T12:41:39.1119020Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_manager.py", line 120 in _hookexec
2025-01-11T12:41:39.1120500Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_hooks.py", line 513 in __call__
2025-01-11T12:41:39.1121700Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/python.py", line 1627 in runtest
2025-01-11T12:41:39.1123010Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/runner.py", line 174 in pytest_runtest_call
2025-01-11T12:41:39.1124360Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_callers.py", line 103 in _multicall
2025-01-11T12:41:39.1125540Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_manager.py", line 120 in _hookexec
2025-01-11T12:41:39.1126740Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_hooks.py", line 513 in __call__
2025-01-11T12:41:39.1127980Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/runner.py", line 242 in <lambda>
2025-01-11T12:41:39.1129510Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/runner.py", line 341 in from_call
2025-01-11T12:41:39.1130750Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/runner.py", line 241 in call_and_report
2025-01-11T12:41:39.1132000Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/runner.py", line 132 in runtestprotocol
2025-01-11T12:41:39.1133300Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/runner.py", line 113 in pytest_runtest_protocol
2025-01-11T12:41:39.1134460Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_callers.py", line 103 in _multicall
2025-01-11T12:41:39.1135880Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_manager.py", line 120 in _hookexec
2025-01-11T12:41:39.1137160Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_hooks.py", line 513 in __call__
2025-01-11T12:41:39.1138430Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/main.py", line 362 in pytest_runtestloop
2025-01-11T12:41:39.1140030Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_callers.py", line 103 in _multicall
2025-01-11T12:41:39.1141470Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_manager.py", line 120 in _hookexec
2025-01-11T12:41:39.1142780Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_hooks.py", line 513 in __call__
2025-01-11T12:41:39.1143940Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/main.py", line 337 in _main
2025-01-11T12:41:39.1145100Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/main.py", line 283 in wrap_session
2025-01-11T12:41:39.1146350Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/main.py", line 330 in pytest_cmdline_main
2025-01-11T12:41:39.1147700Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_callers.py", line 103 in _multicall
2025-01-11T12:41:39.1148960Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_manager.py", line 120 in _hookexec
2025-01-11T12:41:39.1150110Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/pluggy/_hooks.py", line 513 in __call__
2025-01-11T12:41:39.1151330Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/config/__init__.py", line 175 in main
2025-01-11T12:41:39.1153400Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/lib/python3.13t/site-packages/_pytest/config/__init__.py", line 201 in console_main
2025-01-11T12:41:39.1155450Z   File "/private/var/folders/ml/x_tvfgn50_s7p67dm1ypcqqm0000gn/T/cibw-run-kzmap_12/cp313t-macosx_arm64/venv-test-arm64/bin/pytest", line 8 in <module>
2025-01-11T12:41:39.1155920Z 
2025-01-11T12:41:39.1156050Z Extension modules: gmp (total: 1)
2025-01-11T12:41:39.2803320Z ##[endgroup]
2025-01-11T12:41:39.2805300Z                                                             [31m✕ [0m266.82s
2025-01-11T12:41:39.2838040Z ##[error]Command ['/bin/sh', '-c', 'pytest /Users/runner/work/python-gmp/python-gmp/tests/'] failed with code -11. 

```

--- a/main.c
+++ b/main.c
@@ -76,7 +76,7 @@ typedef struct _mpzobject {
 
 PyTypeObject MPZ_Type;
 
-#if !defined(PYPY_VERSION)
+#if !defined(PYPY_VERSION) && !Py_GIL_DISABLED
 #  define CACHE_SIZE (99)
 #else
 #  define CACHE_SIZE (0)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -55,7 +55,7 @@ version = {attr = "setuptools_scm.get_version"}
 "*" = ["*.h"]
 
 [project.optional-dependencies]
-tests = ["pytest", "hypothesis<=6.123.2"]
+tests = ["pytest", "hypothesis<=6.123.4"]
 docs = ["sphinx"]
 develop = ["python-gmp[tests,docs]", "pre-commit", "pyperf",
            "gmpy2>=2.2; platform_python_implementation!='PyPy'",

